### PR TITLE
[SPARK-7518][ML] Tests for comparing probability and prediction in binary logistic regression against R

### PR DESCRIPTION
**Pull request by dbtsai**
Adding more tests for comparing probability and prediction in binary logistic regression against R

### CODE REVIEW

**Comment by dbtsai** [comment]
@mengxr @jkbradley @srowen This is relatively simple change in test. Will help me to have confidence to refactor for MLOR in ML.

File: mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala [hunk 1]
```diff
@@ -488,10 +488,19 @@ private class LogisticAggregator(
     this
   }
 
+  /**
+   * @return The numbers of training samples.
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala]: Though this is a private class, the doc should be `Returns the numbers of training samples.` in order to show up automatically as the method summary in the generated doc. Users have to make one click to see `return ...`

File: mllib/src/test/scala/org/apache/spark/ml/classification/LogisticRegressionSuite.scala [hunk 2]
```diff
@@ -234,11 +236,22 @@ class LogisticRegressionSuite extends FunSuite with MLlibTestSparkContext {
     assert(model.weights(1) ~== weightsR(1) relTol 1E-3)
     assert(model.weights(2) ~== weightsR(2) relTol 1E-3)
     assert(model.weights(3) ~== weightsR(3) relTol 1E-3)
+
+    val result = model.transform(binaryDataset)
```
Comment (mengxr) [inline mllib/src/test/scala/org/apache/spark/ml/classification/LogisticRegressionSuite.scala]: Should the block be a function that takes `model`, `weightsR`, and `binaryDataset`? It is used 9 times in this PR.

File: mllib/src/test/scala/org/apache/spark/ml/classification/LogisticRegressionSuite.scala [hunk 3]
```diff
@@ -234,11 +236,22 @@ class LogisticRegressionSuite extends FunSuite with MLlibTestSparkContext {
...
+    result.select("features", "prediction", "probability").collect().foreach {
+      case Row(features: DenseVector, prediction: Double, probability: Vector) =>
+        val probabilityR = Array.ofDim[Double](2)
+        probabilityR(0) = 1.0 / (1 + Math.exp(dot(features, Vectors.dense(weightsR)) + interceptR))
```
Comment (mengxr) [inline mllib/src/test/scala/org/apache/spark/ml/classification/LogisticRegressionSuite.scala]: `Math` -> `math`. `Math` is deprecated.

File: mllib/src/test/scala/org/apache/spark/ml/classification/LogisticRegressionSuite.scala [hunk 4]
```diff
@@ -266,12 +279,31 @@ class LogisticRegressionSuite extends FunSuite with MLlibTestSparkContext {
...
+        probabilityR(1) = 1.0 - probabilityR(0)
+        assert(probability ~== Vectors.dense(probabilityR) relTol 2E-2)
+        if (prediction != probabilityR.zipWithIndex.maxBy(_._1)._2) {
+          print(prediction)
```
Comment (mengxr) [inline mllib/src/test/scala/org/apache/spark/ml/classification/LogisticRegressionSuite.scala]: We cannot have `print` in the test.

**Pull request closed without merging**